internal/util/update/update.go

```diff
@@ -163,6 +164,15 @@ func (u Command) Run(ctx context.Context) error {
 			}
 
 			if subKf.Upstream != nil && subKf.Upstream.Git != nil {
+				// update subpackage kf ref/strategy if current pkg is a subpkg of root pkg or is root pkg
+				// and if original root pkg ref matches the subpkg ref
+				if isPkgInRootPkg(subKf, rootKf) && subKf.Upstream.Git.Ref == originalRootKfRef {
```

Comment:
This name is confusing. RootPkg means the package on local on which command is being invoked. You should club these two conditions in to single method with name `shouldUpdateSubPkgRef`

Reply:
done